verify_invalid_coupon_error.py

```diff
@@ -0,0 +1,53 @@
+'''
+Write a selenium script that verifies invalid coupon code will show error in cart page
+'''
+
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+
+
+# Keeps webdriver from closes Chrome immediately
+options = webdriver.ChromeOptions()
+options.add_experimental_option('detach', True)
+
+driver = webdriver.Chrome(options=options)
+# driver = webdriver.Firefox()
+
+# Go to website home page
+driver.get('http://frenchy-petes.local/')
+
+
+# Find a product and go to product page
+product_css = 'a.woocommerce-LoopProduct-link.woocommerce-loop-product__link'
+product = driver.find_element(By.CSS_SELECTOR, product_css)
+product.click()
+
+
+# Find add to cart button and click it
+add_to_cart = 'button[value="319"]'
```

Comment:
see now next time you run it the product you click might not be product id 319. 
I think the value=319 is the product id. 
If your site has new products on the first page then your code might break.
Be careful when choosing locators. Always think 'how easily will this locator break/change'. 
I if I am wrong and 319 is the same in all product pages then you are good to go.

Reply:
You're right.  I deleted the product, and now my script does click on the product that is now in the location of the deleted one, but when it goes to the product page it doesn't do anything.
I changed the selector to 'button[name="add-to-cart"]' and now it works fine.